tpm2/tpm2.go

```diff
@@ -1491,3 +1491,92 @@ func encryptDecryptSymmetric(rw io.ReadWriteCloser, keyAuth string, key tpmutil.
 
 	return out, nil
 }
+
+func encodeRSAEncrypt(key tpmutil.Handle, message []byte, scheme *AsymScheme, label string) ([]byte, error) {
+	ha, err := tpmutil.Pack(key)
+	if err != nil {
+		return nil, err
+	}
+	m, err := tpmutil.Pack(message)
+	if err != nil {
+		return nil, err
+	}
+	s, err := scheme.encode()
+	if err != nil {
+		return nil, err
+	}
+	l, err := tpmutil.Pack([]byte(label))
+	if err != nil {
+		return nil, err
+	}
+	return concat(ha, m, s, l)
+}
+
+func decodeRSAEncrypt(resp []byte) ([]byte, error) {
+	var out []byte
+	_, err := tpmutil.Unpack(resp, &out)
+	return out, err
+}
+
+// RSAEncrypt performs RSA encryption in the TPM according to RFC 3447. The key must be
+// a (public) key loaded into the TPM beforehand. Note that when using OAEP with a label,
+// the label needs to be null-terminated and the null byte is included in the padding
```

Comment:
We should null-terminate the string internally

Reply:
Done, only if a label is provided, since using just a null as label changes the results. And adding an extra null to a string that's already null-terminated doesn't have any impact (the TPM uses the first null it finds as per sped).